include/poggles/program.h

```diff
@@ -29,6 +29,7 @@ class POGGLES_EXPORT shader_link_exception : public std::runtime_error
 class POGGLES_EXPORT program
 {
 public:
+  program();
```

Comment:
When I originally wrote this class, I explicitly left out the default ctor. I suppose I should have used the `delete` keyword to indicate this. 

The reason I did so is because I didn't want to deal with what an "empty" program means. Especially since we couldn't use an empty program at the time. 

I see now that you're doing this because you need an empty program with which to use the `compileProgram` function below, but right now that means you can have an empty and useless program exist in your code.

Reply:
Empty constructor gone in favor of initializer list. Soon everything will be gone